File: backend/app/schemas/menu.py
from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel


class MenuNode(BaseModel):
    """단일 메뉴 노드 (트리 응답용)"""

    id: int
    code: str
    name: str
    path: Optional[str] = None
    icon: Optional[str] = None
    sort_order: int = 0
    children: List[MenuNode] = []


class MenuTreeResponse(BaseModel):
    """메뉴 트리 전체 응답"""

    menus: List[MenuNode]

File: frontend/src/types/menu.ts
export type MenuNode = {
  id: number;
  code: string;
  name: string;
  path: string | null;
  icon: string | null;
  sort_order: number;
  children: MenuNode[];
};

export type MenuTreeResponse = {
  menus: MenuNode[];
};

export type MenuActionPermissionItem = {
  action_code: string;
  allowed: boolean;
};

export type MenuActionPermissionResponse = {
  menu_code: string;
  path: string | null;
  allowed_actions: string[];
  actions: MenuActionPermissionItem[];
};
